Gds/src/fprime_gds/common/loaders/xml_loader.py

```diff
@@ -116,8 +117,14 @@ def get_xml_tree(self, path):
 
         # Parse xml and get element tree object we can retrieve data from
         element_tree = etree.parse(fd, parser=xml_parser)
+        root = element_tree.getroot()
 
-        return element_tree.getroot()
+        # Check version of the XML before continuing. Versions weren't published before 1.5.4.  Only check major minor
+        # and point versions to allow for development versions not "bumped" past a point version.
+        dict_version = root.attrib.get("framework_version", "1.5.4")
+        if dict_version < MINIMUM_SUPPORTED_FRAMEWORK_VERSION or dict_version[:5] > MAXIMUM_SUPPORTED_FRAMEWORK_VERSION:
```

Comment:
I'm a little concerned about using a string comparison to compare a semver version number. Are we sure this won't cause issues if we decide to do a release like `2.10.0`?

Reply:
Excellent catch.  I can fix this.